lms/djangoapps/support/views/course_reset.py

```diff
@@ -118,67 +119,48 @@ def post(self, request, username_or_email):
             'course_id': <course id>
             'display_name': <course display name>
             'status': <status of the enrollment wrt/reset, to be displayed to user>
+            'comment': <optional comment made by support staff performing reset>
             'can_reset': (boolean) <can the course be reset for this learner>
         }
         """
-        comment = request.data.get('comment', '')
-        course_id = request.data['course_id']
         try:
+            course_id = request.data['course_id']
+            course_key = CourseKey.from_string(course_id)
             user = get_user_by_username_or_email(username_or_email)
+            opt_in_course = CourseResetCourseOptIn.objects.get(course_id=course_key, active=True)
+        except KeyError:
+            return Response({'error': 'Must specify course id'}, status=400)
+        except InvalidKeyError:
+            return Response({'error': 'invalid course id'}, status=400)
         except User.DoesNotExist:
             return Response({'error': 'User does not exist'}, status=404)
-        try:
-            opt_in_course = CourseResetCourseOptIn.objects.get(course_id=course_id)
         except CourseResetCourseOptIn.DoesNotExist:
             return Response({'error': 'Course is not eligible'}, status=404)
-        enrollment = CourseEnrollment.objects.get(
-            course=course_id,
-            user=user,
-            is_active=True
+
+        if not CourseEnrollment.is_enrolled(user, course_key):
+            return Response({'error': 'Learner is not enrolled in course'}, status=404)
+        course_enrollment = CourseEnrollment.get_enrollment(user, course_key)
+
+        can_reset, status_message = can_enrollment_be_reset(course_enrollment)
+        course_reset_audit = get_latest_audit(course_enrollment)
```

Comment:
Is this needed? Can't see where it is being used as this variable is overridden on line 150 https://github.com/openedx/edx-platform/pull/34419/files#diff-8832436658ef37ffa54d09476ff47d3aabe538b288fb474155aa4909cc080c10R150

Reply:
You're right, that was copy and pasted